### PR TITLE
Feat: Organization system invitation, creation & team creations / switch / join

### LINKED ISSUE #778: feat: create a team, when user switch to a new season
For now if the user did not have a team in the season, he could not join the season.
I want that when a user join a new season, a team is create per default, in a way that a user could join any public season

### LINKED ISSUE #781: feat: choose to deploy a challenge only in a specific season or in all
I don't think there is a easy way to deploy a challenge in a specific season.

### LINKED ISSUE #815: [IDEA] Create page Organization which allows to join / create organization and to see details
a bit like this: 

image

bc as @Mikatech i'm not very skilled w/ front-end, i will just put html classic button without styling, but in a future i will ask someone to render the front better or i will improve my skills w/ front-end :+1:

### PR DESCRIPTION
**Pull request by Mikatech**
I will improve the switch season system & organization system.

I must add:
- [x] We should join the season by slug instead of ID
- [x] Reload data after switching season
- [x] List organizations invitations in web-front-end with three columns (organization, user which invited us, button accept & refuse)
- [x] Allow to create organization in front-end
- [x] Allow to switch to an other season from organization details page
- [x] Find a way to indicate the current season and an intuitiv way to back to global season
- [ ] It would be great if i could notify the user when a new invitation appear (but i don't know how to do this ? i can't just ping every x seconds the route i guess)
- [x] create an activity invitation refused for team & organization

resolve #781 
resolve #778 
resolve #815

### CODE REVIEW

File: go/pkg/pwapi/api_team-get.go [hunk 1]
```diff
@@ -34,11 +34,5 @@ func (svc *service) TeamGet(ctx context.Context, in *TeamGet_Input) (*TeamGet_Ou
 
 	// tmp: fake data
```
Comment (Mikatech) [inline go/pkg/pwapi/api_team-get.go:35]: delete useless comment

File: web/src/actions/organizations.js [hunk 2]
```diff
@@ -1,9 +1,29 @@
 import {
+  ACCEPT_ORGANIZATION_INVITATION_FAILED,
+  ACCEPT_ORGANIZATION_INVITATION_SUCCESS,
+  CREATE_ORGANIZATION_FAILED,
+  CREATE_ORGANIZATION_SUCCESS, DECLINE_ORGANIZATION_INVITATION_FAILED, DECLINE_ORGANIZATION_INVITATION_SUCCESS,
```
Comment (Mikatech) [inline web/src/actions/organizations.js:5]: normalize that

File: web/src/actions/seasons.js [hunk 3]
```diff
@@ -27,25 +27,35 @@ import {
...
-  postPreferences,
   getChallenges,
   getChallengeDetails,
   getTeamDetails,
   postBuyChallenge,
   postValidateChallenge,
   postCloseChallenge,
   postCreateTeam,
+  postPreferencesByID,
+  postPreferencesBySlug, postInviteUserToTeam, postAnswerTeamInvitation,
```
Comment (Mikatech) [inline web/src/actions/seasons.js:51]: normalize that

File: web/src/api/seasons.js [hunk 4]
```diff
@@ -1,10 +1,14 @@
 import { baseApi } from "./index";
 
 //Season main calls
-export function postPreferences(seasonID) {
+export function postPreferencesByID(seasonID) {
   return baseApi.post(`/user/preferences`, { active_season_id: seasonID });
```
Comment (Mikatech) [inline web/src/api/seasons.js:5]: could make one call with getByIdOrSlug in back-end

**Comment by Mikatech** [comment]
Web: add redux state when executing action for (creating organization, creating team)

**Comment by Mikatech** [comment]
Web: Bringing forward the possibility of changing seasons (the user could see the current season from navbar)

**Review by moul — APPROVED** [review]

**Pull request merged**